vim/autoload/himalaya/msg.vim

```diff
@@ -18,21 +18,25 @@ function! s:format_msg_for_list(msg)
   return msg
 endfunction
 
+function! himalaya#msg#list_preview(mbox, page)
+  let msgs = s:cli("--mailbox %s list --page %d", [shellescape(a:mbox), a:page], printf("Fetching %s messages", a:mbox))
+  let msgs = map(msgs, "s:format_msg_for_list(v:val)")
+  let buftype = stridx(bufname("%"), "Himalaya messages") == 0 ? "file" : "edit"
+  execute printf("silent! %s Himalaya messages [%s] [page %d]", buftype, a:mbox, a:page + 1)
+  setlocal modifiable
+  silent execute "%d"
+  call append(0, s:render("list", msgs))
+  silent execute "$d"
+  setlocal filetype=himalaya-msg-list
+  let &modified = 0
+  execute 0
+endfunction
+
 function! himalaya#msg#list()
   try
     let mbox = himalaya#mbox#curr_mbox()
     let page = himalaya#mbox#curr_page()
-    let msgs = s:cli("--mailbox %s list --page %d", [shellescape(mbox), page], printf("Fetching %s messages", mbox))
-    let msgs = map(msgs, "s:format_msg_for_list(v:val)")
-    let buftype = stridx(bufname("%"), "Himalaya messages") == 0 ? "file" : "edit"
-    execute printf("silent! %s Himalaya messages [%s] [page %d]", buftype, mbox, page + 1)
-    setlocal modifiable
-    silent execute "%d"
-    call append(0, s:render("list", msgs))
-    silent execute "$d"
-    setlocal filetype=himalaya-msg-list
-    let &modified = 0
-    execute 0
+    call himalaya#msg#list_preview(mbox, page)
```

Comment:
I did not know you wanted to call the `list_preview` fn here also, and in fact it makes sense. This way you reuse the logic. But the name does not match anymore. Maybe sth like `list_with`? Or whatever you find more appropriate ;)

Reply:
Good point, changes to `list_with`